### PR TITLE
chore: Drop scala java8 compat

### PR DESCRIPTION
**Pull request by sebastian-alfers**

### CODE REVIEW

**Comment by sebastian-alfers** [comment]
Fixes https://github.com/lightbend/akka-meta/issues/440

**Review by octonato — APPROVED** [review]
All looks fine, but my only concern is for users that are using java8Compat because they get it transitively from Akka. 

Maybe not a major problem that should block us though.

**Comment by ennru** [comment]
This should wait until the last responsible moment before the Akka 2.10 version.

**Comment by patriknw** [comment]
We will not make such dependency change in a patch version, but it would be fine for Akka 2.10. Let's wait with merging so that we don't have to maintain double branches, for a while more.

**Comment by patriknw** [comment]
I have created branch `release-2.9`. `main` continues as dev branch for 2.10. This can be continued.

**Review by johanandren — COMMENTED** [review]
Something in the release notes about scala-java8-compat not being a transitive dependency anymore would be good.

**Review by johanandren — APPROVED** [review]

File: akka-docs/src/main/paradox/project/migration-guide-2.9.x-2.10.x.md [hunk 1]
```diff
@@ -6,3 +6,7 @@ project.description: Migrating to Akka 2.9.
 ## Support for slf4j 1.7.x and logback 1.2.x removed
 
 This is the first release that only supports slf4j 2.0.x and logback 1.5.x.
+
+## Remove dependency to scala-java8-compat
+
+The transitive dependency on `scala-java8-compat` has been removed.
```
Comment (patriknw) [inline akka-docs/src/main/paradox/project/migration-guide-2.9.x-2.10.x.md:12]: 👍

File: akka-actor-tests/src/test/java/akka/pattern/CircuitBreakerTest.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ public void useCircuitBreakerWithCompletableFuture() throws Exception {
     assertEquals(
         "hello",
         Await.result(
-            FutureConverters.toScala(res), JavaDurationConverters.asFiniteDuration(fiveSeconds)));
+            FutureConverters.asScala(res), JavaDurationConverters.asFiniteDuration(fiveSeconds)));
```
Comment (patriknw) [inline akka-actor-tests/src/test/java/akka/pattern/CircuitBreakerTest.java:46]: In a separate PR you could try to remove `akka.util.JavaDurationConverters`

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**